ttnn/cpp/ttnn/operations/data_movement/concat/device/concat_device_operation.cpp

```diff
@@ -153,13 +154,13 @@ Tensor concat_impl(std::vector<Tensor> &input_tensors, const std::int64_t dim, c
                 if (input_tensors[0].get_layout() == Layout::ROW_MAJOR && normalized_dim == ref_rank - 1) {
                     for (const auto &input_tensor : input_tensors) {
                         TT_FATAL(
-                            (input_tensor.get_legacy_shape()[dim] * input_tensor.element_size()) % input_tensor.buffer()->alignment() ==
+                            (input_tensor.get_padded_shape()[dim] * input_tensor.element_size()) % input_tensor.buffer()->alignment() ==
                                 0,
                             "Current concat implementation requires aligned last dim when concatting on last dim");
                     }
                 }
                 // row major should default to row major and tilized to tilized implementations, but the below loop turned RM to tilized when possible
-                Layout target_layout = input_tensors[0].get_layout();
+                Layout target_layout = Layout::TILE;
```

Comment:
default to tile? Is this preferred?

Reply:
I think I brought this back during a rebase onto your concat work; thanks for catching it. I'll make sure that it still works without this diff.

Reply:
Reverting this results in hitting a hang in `ttnn::tilize` for the tiled cases that need to be converted to row major ☹️ Will have to take a closer look at tilize.

Reply:
maybe that could cause the perf issue? Do you know if there's a concat in that model that takes the input as RM?